RELEASE_PROCEDURE.md

```diff
@@ -0,0 +1,11 @@
+# Release procedure
+
+- Update [version_history.md](docs/root/version_history.md). 
+  - Make sure breaking changes are explicitly called out.
```

Comment:
Ideally we do these as we go, rather than a major effort at release time. The experience form Envoy has been that this scales better.

Reply:
I'll make sure we mention that here - though maybe we could also have a pre-merge reviewer checklist that captures this; when someone about to perform a release read it here it's probably too late?

Reply:
Sure. Could be useful for new maintainers, +1

Reply:
> Sure. Could be useful for new maintainers, +1

Filed https://github.com/envoyproxy/nighthawk/issues/230, let me know if you have any more things that should go in there.